src/store/activities/actions.js

```diff
@@ -25,8 +25,45 @@ export const removeActivity = id => ({
   id,
 });
 
+export const startRemoveActivity = id => async dispatch => {
+  try {
+    await database.ref(`activities/items/${id}`).remove();
+    return dispatch(removeActivity(id));
+  } catch (error) {
+    return console.error(`Ops! ${error}`); // eslint-disable-line no-console
+  }
+};
+
 export const editActivity = (id, updates) => ({
   type: EDIT_ACTIVITY,
   id,
   updates,
 });
+
+export const startEditActivity = (id, updates) => async dispatch => {
+  try {
+    await database.ref(`activities/items/${id}`).update(updates);
+    return dispatch(editActivity(id, updates));
+  } catch (error) {
+    return console.error(`Ops! ${error}`); // eslint-disable-line no-console
+  }
+};
+
+export const setActivities = activities => ({
+  type: SET_ACTIVITIES,
+  activities,
+});
+
+export const startSetActivities = () => async dispatch => {
+  try {
+    const snapshot = await database.ref('activities/items').once('value');
+    const fetchedData = snapshot.val();
+    const activities = [];
+    Object.keys(fetchedData).map(item => {
```

Comment:
I don't like this. I think that there are some methods in firebase to loop and change values, we shouldn't make such a hack.